src/api/routes/author.js

```diff
@@ -37,17 +37,155 @@ const authorBasicRelations = [
 const authorError = 'Author not found';
 
 /**
- * @swagger
- *
- * /author:
- *   get:
- *     description: Retrive the basic information of an author
+ *@swagger
+ *definitions:
+ *  defaultAlias:
+ *    type: object
+ *    properties:
+ *      name:
+ *        type: string
+ *        example: 'Robert A. Heinlein'
+ *      sortName:
+ *        type: string
+ *        example: 'Heinlein, Robert A.'
+ *      aliasLanguage:
+ *        type: string
+ *        example: 'English'
+ *  AuthorDetail:
+ *   type: object
+ *   properties:
+ *     bbid:
+ *       type: string
+ *       format: uuid
+ *       example: '2e5f49a8-6a38-4cc7-97c7-8e624e1fc2c1'
+ *     beginArea:
+ *       type: string
+ *       example: 'United States'
+ *     beginDate:
+ *       type: string
+ *       example: '1907-07-07'
+ *     defaultAlias:
+ *         $ref: '#/definitions/defaultAlias'
+ *     disambiguation:
+ *       type: string
+ *       example: 'Robert A. Heinlein'
+ *     endArea:
+ *       type: string
+ *       example: 'United States'
+ *     endDate:
+ *       type: string
+ *       example: '1988-05-08'
+ *     ended:
+ *       type: boolean
+ *       example: true
+ *     gender:
+ *       type: string
+ *       example: 'Male'
+ *     type:
+ *       type: string
+ *       example: 'Person'
+ *  AuthorAliases:
```

Comment:
Shouldn't AuthorAliases, AuthorIdentifiers and AuthorRelationships be generic `Aliases`, `Identifiers` and `Relationships`, common for all entity types?
It should be the same as `defaultAlias`, the same for all entities.

Reply:
I'm not sure where these common definitions should live (same for defaultAlias).
Perhaps in the `src/api/routes.js` file? In which case you'll need to add that path to the `apis` array in the swagger options (swagger.js file)

Reply:
We can put these definitions in one external file but the problem I found, How to pass entity data to these definitions and how to make the variable key-value of these definitions to use for all entity.

Reply:
@akhilesh26 I'm not sure I understand what's troubling you.
I tried locally to rename and move the aliases, relationships and identifiers swagger definitions to src/api/routes.js , and modified api/routes/author.js to point to them (`$ref: '#/definitions/aliases` instead of `$ref: '#/definitions/AuthorAliases`).
That's working fine for me:
<img width="1198" alt="Capture d’écran 2019-07-29 à 12 23 25" src="https://user-images.githubusercontent.com/6179856/62041529-29a40200-b1fc-11e9-8180-66403ab94f48.png">
![Swagger UI - author](https://user-images.githubusercontent.com/6179856/62041540-2dd01f80-b1fc-11e9-8d4b-fa0b3da0a975.png)



If you are concerned with having an aliases example with an author's name when looking at other entities, I don't think that's a problem, as they are just meant as examples and what matters most is the expected structure.